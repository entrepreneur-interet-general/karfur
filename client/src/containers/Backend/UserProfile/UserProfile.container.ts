import { compose } from "recompose";
import {
  PropsBeforeInjection,
  UserProfileComponent,
} from "./UserProfile.component";
import { withTranslation, WithTranslation } from "react-i18next";

interface InjectedProps extends WithTranslation {}
export interface Props extends PropsBeforeInjection, InjectedProps {}

export const UserProfileContainer = compose<Props, PropsBeforeInjection>(
  withTranslation()
)(UserProfileComponent);
