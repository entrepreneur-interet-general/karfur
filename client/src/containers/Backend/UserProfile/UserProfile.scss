@import 'src/scss/global.scss';
@import 'src/scss/colors.scss';

$defaultSize:310px;
.user-profile{
  background: $blanc;
  min-height:100%;
  font-size: 18px;
  line-height: 23px;
  padding-top:100px;
  flex: 1 1;

  .profile-header{
    height: 60px;
    width:100%;
    display:flex;
    flex-direction: row;
    justify-content: center;
    align-items: center;
    position: fixed;
    z-index: 3;
    background: $gris;
    top: 60px;

    a{
      font-style: normal;
      font-weight: bold;
      color: $noir;
      padding:18px 25px;
      &:hover{
        background-color: $noir;
        color: $blancSimple;
        svg{
          fill: #FFFFFF;
        }
        .svgico{
          filter: invert(100%);
        }
    
      }

      .header-icon{
        margin-right:8px;
      }
    }
    
  }

  .profile-content{
    margin: 23px 60px;
    @include media-max("phone-down") { margin: 23px 32px; }

    .profile-info{
      background: inherit;
      border-radius: 12px;
      max-width: 100%;
      margin:0;
      
      .modify-col{
        padding: 0 10px;
      }

      .user-col {
        padding-left: 0px;
        padding-right: 0px;
      }

      .profile-left{
        text-align: center;
        width: 200px;

        .card-body{
          padding:20px;
          padding-bottom:10px;
          border-radius: 12px 12px 0px 0px;
          background: $gris;

          .user-picture{
            width: 120px;
          }
        }
        .card-footer{
          background: $gris;
          border-radius: 0px 0px 12px 12px;
          color: $noir;
          text-align: center;
          border:none;
          padding:0 20px 20px 20px;

          .name{
            font-weight: bold;
            font-size: 18px;
            line-height: 23px;
            margin:0;
          }
        }

        .profile-header-container{
          width: 100%;
        }
      
        .profile-header-img > img.img-circle {
          width: 100%;
          height: 100%;
          border: 2px solid #51D2B7;
        }
      
        .profile-header {
          margin-top: 43px;
        }
      
        .rank-label-container {
          text-align: center;
          min-height:120px;
          overflow: hidden;
          position: relative;
          display: flex;
          flex-direction: column;
          justify-content: center;
          align-items: center;
        }
      
        .label.label-default.rank-label {
          background-color: rgb(81, 210, 183);
          margin-top:-19px;
        }
      }
      
      .align-right{
        display: inline-flex;
        float: right;
      }
      .margin-right-8{
        margin-right:8px;
      }
      .pointer{
        cursor: pointer;
      }

      .shadow-wrapper{
        border-radius: 12px;
        height: 100%;
        display: flex;
        flex-direction: column;
        &.active{
          box-shadow: 0px 10px 15px rgba(0, 0, 0, 0.25);
        }
      }

      .profile-modify{
        border-radius: 12px;
        color: $noir;
        border:none;
        background: transparent;
        height: 100%;

        .shadow-wrapper{
          background: $gris;
          padding: 20px;
          height: 100%;
        }
        .card-body{
          border-radius: 0px 0px 12px 12px;
          border:none;
          padding: 0;

          .row{
            margin-bottom:23px;
          }
          div[contenteditable="true"]{
            background: white;
            border-radius: 6px;
            min-height: 30px;
            &:after{
              border-radius: 6px;
            }
            &:empty{
              min-height: 30px;
            }
            .input{
              text-align: center;
            }
          }
          .langues-dropdown .btn, .plus-langue{
            background: linear-gradient(110.29deg, #FBFBFB 0%, #F2F2F2 100%);
            border-radius: 12px;
            margin-right:10px;
            border:none;
            font-weight: bold;
            font-size: 14px;
            line-height: 18px;
            color: $noir;
            padding: 6px 10px;
          }
          .minus-icon{
            position: absolute;
            right: -7px;
            top: -15px;
            cursor: pointer;
          }

          .handleBetween {
            font-weight: bold;
          }
          @include media-between("xl-limit", "widescreen-up") { 
            .handleBetween{
              flex: 0 0 40%;
              max-width: 40%;
              overflow-wrap: initial;
            }
          }
        }
        .card-footer{
          background: transparent;
          border:none;
          padding:0;
          display: flex;
          justify-content: flex-end;

          .btn{
            margin-left:10px;
          }
          .validate-btn{
            float:right;
          }
        }

        .nb-car-row{
          margin-top:-20px;
        }
        .texte-vert{
          color: $validationHover;
        }
      }

      .profile-right{
        border:none;
        background: inherit;
        height: 100%;
        
        a{
          display: inherit;
          flex-direction: inherit;
          justify-content: inherit;
          height: 100%;
        }
        .card-body{
          padding:0;
          height:100%;
          margin-bottom: none;
          .row{
            margin: 0 -10px 0 0;
            height: 100%;
            .col{
              padding: 20px 20px;
              background: $gris;
              border-radius: 12px;
              margin-right: 10px;
              width: 200px;
              max-width: 200px;
              height: 100%;
              line-height: 20px;
              display: flex;
              flex-direction: column;
              color: $noirCD;
              h1{
                margin-bottom:10px;
              }
              .subtitle{
                margin-bottom: 30px;
              }
              .content{
                color: $noir;
                margin-top: auto;

              }
              &.active{
                color: $noir;
              }
            }
          }
        }
        .card-footer{
          background: rgba(185, 236, 255, 0.3);
          border-radius: 12px;
          margin-top: 10px;
          border:none;
          color: #60A3BC;
        }
      }
    }
  }
  
  .btn{
    border-radius: 12px;
  }
  .tag-btn{
    margin-right:10px;
    color: #FFFFFF;
  }

  @include media-max("desktop-up") { 
    .hideOnPhone{
      display:none;
    }
  }


  .tableau-wrapper .tableau{
    background: $gris;
  }
}

.modal.show .modal-dialog.modal-plus{
  overflow: scroll;
  max-height: 90vh;
  max-width: 60vw;
  margin: auto;
  overflow: scroll;
  @include media-max("tablet-up") { max-width: 100%; }

  .modal-content{
    padding:20px;
    max-height: 90vh;
  }
}

.avancement-user-table.table, .avancement-table .table{
  border-style : hidden;
  thead{
    font-weight: normal;
    font-size: 16px;
    line-height: 20px;
    color:$noir;
  }
  tbody tr:hover{
    background: $blanc;
    cursor: pointer;
    border-style: hidden;
    border-radius: 12px;
    &.terminee{
      background: inherit;
      cursor: inherit;
    }
  }
  td {
    border-style : hidden;
    font-style: normal;
    font-weight: normal;
    font-size: 18px;
    line-height: 23px;
    color: $noir;
  }
  .voir-plus{
    text-align: center;
    cursor: pointer;
  }
  .dot-circle{
    width: 13px;
    height: 13px;
    border-radius: 50%;
    background: $noir;
    box-shadow: 0px 0px 16px rgba(0, 0, 0, 0.3);
  }
  .depuis{
    &.alert{
      color: $rouge;
    }
    &.success{
      color: $vert;
    }
  }
  .fit-content{
    width: 1%;
  }
  .status-pill{
    width: max-content;
    padding: 6px 10px;
    border-radius: 6px;
  }
}

.tableau-header{
  width:100%;
  padding:0;
  font-style: normal;
  margin-bottom:30px;
  min-width: fit-content;
  &.no-margin{
    margin:0;
  }
  .pointer{
    cursor: pointer;
  }

  h4{
    font-size: 42px;
    line-height: 53px;
    margin: auto 23px auto 0;
    display: inline;
    margin:0;
  }

  .left-element, .middle-element{
    font-style: normal;
    font-weight: bold;
    align-items: center;
  }

  .equi-reparti{
    display: flex;
    justify-content: space-between;
  }

  .ml-75{
    margin-left:75px;
  }

  .right-element{
    display: flex;
    align-items: center;
    .modify-obj{
      font-size: 18px;
      color: #828282;
    }
  }

  .full-width{
    width: 100%;
  }
  &.align-right{
    display: flex;
    justify-content: flex-end;
  }
}

.user-profile, .user-dash-contrib, .user-dash, .user-dash-struct{
  .tableau-wrapper{
    margin-top:39px;
    width: 100%;
    position: relative;
    
    .tableau{
      position: relative;
    }

    &.swing-out-top-bck {
      -webkit-animation: swing-out-top-bck 1s cubic-bezier(0.600, -0.280, 0.735, 0.045) both;
              animation: swing-out-top-bck 1s cubic-bezier(0.600, -0.280, 0.735, 0.045) both;
    }

    h1{
      font-size: 40px;
      display: inline;
      @include media-max("phone-down") { font-size: 46px; }

      .sup, .nb-actions{
        display: inline-block;
        width: 1.5em;
        height: 1.5em;
        font-weight: bold;
        text-align: center;
        font-size: 18px;
        line-height: 1.5em;
        background-color: $noir;
        border-radius: 1000px;
        color: $blanc;
        top: -25px;
        left: 10px;
      }
    }

 

    .tableau{
      border-radius: 12px;
      .progress{
        height:6px;
        margin-top: 10px;
        margin-bottom: 0;
      }
      .langue-item {
        i{
          margin-right:8px;
        }
      }
    }
    .ecran-protection{
      .content-wrapper{
        max-width: 650px;
      }
    }
    .tableau-footer{
      display: flex;
      justify-content: flex-end;

      .btn{
        background: linear-gradient(147.25deg, #4A69BD -7.66%, #60A3BC 92.45%);
        box-shadow: 0px 0px 16px rgba(0, 0, 0, 0.3);
        border-radius: 12px;
        span{
          font-weight: bold;
          font-size: 18px;
          line-height: 23px;
          margin-left:8px;
        }
      }
    }

    .pointer{
      cursor: pointer;
    }
  }
}

@-webkit-keyframes swing-out-top-bck {
  0% {
    -webkit-transform: rotateX(0deg);
            transform: rotateX(0deg);
    -webkit-transform-origin: top;
            transform-origin: top;
    opacity: 1;
    height:100%;
  }
  90%{
    height:100%;
  }
  100% {
    -webkit-transform: rotateX(-100deg);
            transform: rotateX(-100deg);
    -webkit-transform-origin: top;
            transform-origin: top;
    opacity: 0;
    height:0;
    margin:0;
  }
}
@keyframes swing-out-top-bck {
  0% {
    -webkit-transform: rotateX(0deg);
            transform: rotateX(0deg);
    -webkit-transform-origin: top;
            transform-origin: top;
    opacity: 1;
    height:100%;
  }
  90%{
    height:100%;
  }
  100% {
    -webkit-transform: rotateX(-100deg);
            transform: rotateX(-100deg);
    -webkit-transform-origin: top;
            transform-origin: top;
    opacity: 0;
    height:0;
    margin:0;
  }
}
