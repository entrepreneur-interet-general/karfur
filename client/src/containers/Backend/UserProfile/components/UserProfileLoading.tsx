import React from "react";
import {
  MainContainer,
  ProfilePictureContainer,
  ProfileContainer,
  Title,
} from "../UserProfile.component";
import styled from "styled-components";
import Skeleton, { SkeletonTheme } from "react-loading-skeleton";

const ImageContainer = styled.div`
  width: 160px;
  height: 160px;
  border-radius: 50%;
  background-color: #ffffff;
  margin-bottom: 8px;
`;
export const UserProfileLoading = (props: { t: any }) => (
  <MainContainer>
    <ProfilePictureContainer>
      <ImageContainer />
      <SkeletonTheme color="#FFFFFF">
        <Skeleton count={1} height={20} width={160} />
      </SkeletonTheme>
    </ProfilePictureContainer>
    <ProfileContainer>
      <Title>{props.t("UserProfile.votre pseudo", "Votre pseudonyme")}</Title>
      <SkeletonTheme color="#FFFFFF">
        <Skeleton count={1} height={50} />
      </SkeletonTheme>
      <Title marginTop={"24px"}>
        {props.t("Register.Votre email", "Votre email")}
      </Title>
      <SkeletonTheme color="#FFFFFF">
        <Skeleton count={1} height={50} />
      </SkeletonTheme>

      <Title marginTop={"24px"}>
        {props.t("UserProfile.Votre mot de passe", "Votre mot de passe")}
      </Title>
      <SkeletonTheme color="#FFFFFF">
        <Skeleton count={1} height={50} />
      </SkeletonTheme>
    </ProfileContainer>
  </MainContainer>
);
