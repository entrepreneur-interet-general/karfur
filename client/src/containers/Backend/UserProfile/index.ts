export { UserProfileContainer as UserProfile } from "./UserProfile.container";
