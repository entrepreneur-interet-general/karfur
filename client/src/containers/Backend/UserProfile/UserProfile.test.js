import React from 'react';
import {UserProfile, PasswordModal} from './UserProfile';
import { expect } from 'chai';
import { shallow } from 'enzyme';
import configureStore from 'redux-mock-store';
import { Provider } from 'react-redux';
import sinon from 'sinon';
import axios from 'axios';
import { Modal } from 'reactstrap';
import MongoClient, {ObjectId} from "mongodb";

import { FavoriTable, ContribTable, ActionTable, TradTable } from '../../../components/Backend/UserProfile';
import { ThanksModal, SuggestionModal, TraducteurModal, ObjectifsModal, AddMemberModal } from '../../../components/Modals';

const mockStore = configureStore();
const store = mockStore({});
const mockId = "a0000000000000000000000a"; //"5cbabcdbcabae0fd31cb13e0"
const mockUser = {
  _id: new ObjectId(mockId), //"5cbabcdbcabae0fd31cb13e0",
  "username" : "mockUser",
  "status" : "Inactif",
  "description" : "description",
  "email" : "mockUser@mockUser.fr",
  "cookies" : { },
  "contributions" : [],
  "structures" : [],
  "roles" : [],
  selectedLanguages: [],
};

const defaultProps = {
  t:k=>k,
  user: mockUser,
}

describe('UserProfile', () => {
  let wrapper, connection, db;
  beforeEach(async () => {
    connection = await MongoClient.connect('mongodb://localhost/db', {
      useNewUrlParser: true,
    });
    db = await connection.db("db");
    const users = db.collection('users');

    await users.findOne({_id: new ObjectId(mockId)});
    console.log('removing mockUser')
    await users.remove({_id: new ObjectId(mockId)});
    console.log('mockUser removed')
    await users.findOne({_id: new ObjectId(mockId)});
    console.log('inserting mockUser')
    await users.insert(mockUser);
    console.log('mockUser inserted')
    await users.findOne({_id: new ObjectId(mockId)});

    wrapper = shallow(<Provider store={store}><UserProfile {...defaultProps} /></Provider> ).dive().dive();
  })

  afterAll(async () => {
    const users = db.collection('users');
    await users.remove({_id: new ObjectId(mockId)});
    await connection.close();
    await db.close();
  });

  it('should insert correct test user', async () => {
    const users = db.collection('users');
    const insertedUser = await users.findOne({_id: new ObjectId(mockId)});
    expect(insertedUser).to.deep.equal(mockUser);
  });

  it('renders without crashing', () => {
    expect(wrapper.is('div')).to.equal(true);
    expect(wrapper.find('span.hideOnPhone').at(0).text()).to.equal("Tables.Mon profil");
  });

  it('contains UserProfile in className', () => {
    expect(wrapper.props().className).to.contain("user-profile");
  });

  it('calls componentDidMount', () => {
    sinon.spy(UserProfile.prototype, 'componentDidMount');
    shallow(<Provider store={store}><UserProfile {...defaultProps} /></Provider> ).dive().dive();
    expect(UserProfile.prototype.componentDidMount).to.have.property('callCount', 1);
    UserProfile.prototype.componentDidMount.restore();
  });

  it('should match its reference snapshot', () => {
    expect(wrapper).to.matchSnapshot()
  });

  it('renders props correctly', () => {
    expect(wrapper.find('h2.name').text()).to.equal(defaultProps.user.username);
  });

  it('should have an empty user in state', () => {
    expect(wrapper.state().user).to.be.an('object').that.is.empty;
  });

  it('renders children when passed in', () => {
    const wrapper = shallow((
      <Provider store={store}>
        <UserProfile {...defaultProps} >
          <div className="unique" />
        </UserProfile>
      </Provider>
    ));
    expect(wrapper.contains(<div className="unique" />)).to.equal(true);
  });

  it('should contain all tables', () => {
    expect(wrapper.containsMatchingElement(FavoriTable)).to.equal(true);
    expect(wrapper.containsMatchingElement(ContribTable)).to.equal(true);
    expect(wrapper.containsMatchingElement(TradTable)).to.equal(true);
    expect(wrapper.containsMatchingElement(ActionTable)).to.equal(true);
  });

  it('should have 4 modals', () => {
    expect(wrapper.find(Modal)).to.have.lengthOf(4);
  });

  it('should contain all external modals', () => {
    expect(wrapper.containsMatchingElement(ThanksModal)).to.equal(true);
    expect(wrapper.containsMatchingElement(SuggestionModal)).to.equal(true);
    expect(wrapper.containsMatchingElement(TraducteurModal)).to.equal(true);
    expect(wrapper.containsMatchingElement(ObjectifsModal)).to.equal(true);
    expect(wrapper.containsMatchingElement(AddMemberModal)).to.equal(true);
    expect(wrapper.containsMatchingElement(PasswordModal)).to.equal(true);
  });

  it('all modals should be hidden', () => {
    expect(wrapper.find(ThanksModal).props().show).to.equal(false);
    expect(wrapper.find(SuggestionModal).props().show).to.equal(false);
    expect(wrapper.find(TraducteurModal).props().show).to.equal(false);
    expect(wrapper.find(ObjectifsModal).props().show).to.equal(false);
    expect(wrapper.find(AddMemberModal).props().show).to.equal(false);
    expect(wrapper.find(PasswordModal).props().show).to.equal(false);
  });

  it('should receive users from real API', function (done) {
    axios.interceptors.response.use(response => {
      if(response.config.url.includes('http://localhost:3000/user/get_users')){
        const data = response.data.data;
        expect(data).to.be.an('array').that.is.not.empty;
        expect(data).to.have.lengthOf.above(1);
        expect(data).to.have.nested.property('0._id');
        expect(data).to.have.nested.property('0.username');
        done();
      }
      return response;
    }, error => {
      console.log('Error: ', error.message);  
    })
    shallow(<Provider store={store}><UserProfile {...defaultProps} /></Provider> ).dive().dive();
  })
})

