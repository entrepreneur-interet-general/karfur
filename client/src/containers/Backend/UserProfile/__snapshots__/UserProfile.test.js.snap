// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserProfile should match its reference snapshot 1`] = `
<div
  className="animated fadeIn user-profile"
>
  <div
    className="profile-header"
  >
    <AnchorLink
      className="header-anchor d-inline-flex justify-content-center align-items-center"
      href="#mon-profil"
      offset="120"
    >
      <EVAIcon
        className="header-icon"
        fill="#fff"
        name="settings-2-outline"
        size="medium"
      />
       
       
      <span
        className="hideOnPhone"
      >
        Tables.Mon profil
      </span>
    </AnchorLink>
    <AnchorLink
      className="header-anchor d-inline-flex justify-content-center align-items-center"
      href="#mes-favoris"
      offset="120"
    >
      <EVAIcon
        className="header-icon"
        fill="#fff"
        name="bookmark-outline"
        size="medium"
      />
       
       
      <span
        className="hideOnPhone"
      >
        Tables.Favoris
      </span>
    </AnchorLink>
    <AnchorLink
      className="header-anchor d-inline-flex justify-content-center align-items-center"
      href="#mes-contributions"
      offset="120"
    >
      <EVAIcon
        className="header-icon"
        fill="#fff"
        name="file-add-outline"
        size="medium"
      />
       
       
      <span
        className="hideOnPhone"
      >
        Tables.Rédactions
      </span>
    </AnchorLink>
    <AnchorLink
      className="header-anchor d-inline-flex justify-content-center align-items-center"
      href="#mes-traductions"
      offset="120"
    >
      <SVGIcon
        className="header-icon svgico"
        name="translate"
      />
       
       
      <span
        className="hideOnPhone"
      >
        Tables.Traductions
      </span>
    </AnchorLink>
  </div>
  <div
    className="profile-content"
    id="mon-profil"
  >
    <Row
      className="profile-info"
      tag="div"
    >
      <div
        className="profile-left mt-10"
      >
        <div
          className="shadow-wrapper"
        >
          <CardBody
            tag="div"
          >
            <div
              className="profile-header-container"
            >
              <div
                className="rank-label-container"
              >
                <img
                  alt="profile"
                  className="img-circle user-picture"
                  src="mario-profile.jpg"
                />
              </div>
            </div>
          </CardBody>
          <CardFooter
            tag="div"
          >
            <h2
              className="name"
            >
              mockUser
            </h2>
            <span
              className="status"
            >
              UserProfile.Utilisateur
            </span>
          </CardFooter>
        </div>
      </div>
      <Col
        className="modify-col mt-10"
        tag="div"
        widths={
          Array [
            "xs",
            "sm",
            "md",
            "lg",
            "xl",
          ]
        }
      >
        <withI18nextTranslation(modifyProfile)
          actions={Array []}
          actionsStruct={Array []}
          contributeur={false}
          contributions={Array []}
          contributionsStruct={Array []}
          cpassword=""
          editing={false}
          favoris={Array []}
          handleChange={[Function]}
          isDropdownOpen={Array []}
          isMainLoading={false}
          langues={Array []}
          nbReadStruct={0}
          newPassword=""
          password=""
          passwordVisible={false}
          progression={
            Object {
              "nbMots": 0,
              "nbMotsContrib": 0,
              "timeSpent": 0,
            }
          }
          selected={Object {}}
          showModal={
            Object {
              "actions": false,
              "addMember": false,
              "contributions": false,
              "devenirContributeur": false,
              "devenirTraducteur": false,
              "favori": false,
              "objectifs": false,
              "password": false,
              "suggestion": false,
              "thanks": false,
              "traducteur": false,
            }
          }
          showSections={
            Object {
              "contributions": true,
              "traductions": true,
            }
          }
          structure={Object {}}
          suggestion={Object {}}
          tempImg={null}
          toggleEditing={[Function]}
          toggleModal={[Function]}
          traducteur={false}
          traductions={Array []}
          traductionsStruct={Array []}
          uploading={false}
          user={
            Object {
              "_id": "a0000000000000000000000a",
              "contributions": Array [],
              "cookies": Object {},
              "description": "description",
              "email": "mockUser@mockUser.fr",
              "roles": Array [],
              "selectedLanguages": Array [],
              "status": "Actif",
              "structures": Array [],
              "username": "mockUser",
            }
          }
          users={Array []}
          validateProfile={[Function]}
        />
      </Col>
      <Col
        className="user-col mt-10"
        lg="12"
        md="12"
        sm="12"
        tag="div"
        widths={
          Array [
            "xs",
            "sm",
            "md",
            "lg",
            "xl",
          ]
        }
        xl="auto"
        xs="12"
      >
        <Card
          className="profile-right"
          tag="div"
        >
          <CardBody
            tag="div"
          >
            <Row
              tag="div"
            >
              <Col
                className="obj-col obj-first"
                lg="4"
                md="4"
                sm="12"
                tag="div"
                widths={
                  Array [
                    "xs",
                    "sm",
                    "md",
                    "lg",
                    "xl",
                  ]
                }
                xl="auto"
                xs="12"
              >
                <AnchorLink
                  href="#mes-contributions"
                  offset="120"
                >
                  <h1
                    className="title text-big"
                  >
                    0
                  </h1>
                  <h6
                    className="subtitle"
                  >
                    UserProfile.minutes données
                  </h6>
                  <span
                    className="content texte-small"
                  >
                    UserProfile.commencez à contribuer
                    .
                  </span>
                </AnchorLink>
              </Col>
              <Col
                className="obj-col obj-second"
                lg="4"
                md="4"
                sm="12"
                tag="div"
                widths={
                  Array [
                    "xs",
                    "sm",
                    "md",
                    "lg",
                    "xl",
                  ]
                }
                xl="auto"
                xs="12"
              >
                <AnchorLink
                  href="#mes-contributions"
                  offset="120"
                >
                  <h1
                    className="title text-big"
                  >
                    0
                  </h1>
                  <h6
                    className="subtitle"
                  >
                    UserProfile.mots écrits
                  </h6>
                  <span
                    className="content texte-small"
                  >
                    UserProfile.commencez à rédiger
                    .
                  </span>
                </AnchorLink>
              </Col>
              <Col
                className="obj-col obj-third"
                lg="4"
                md="4"
                sm="12"
                tag="div"
                widths={
                  Array [
                    "xs",
                    "sm",
                    "md",
                    "lg",
                    "xl",
                  ]
                }
                xl="auto"
                xs="12"
              >
                <AnchorLink
                  href="#mes-traductions"
                  offset="120"
                >
                  <h1
                    className="title text-big"
                  >
                    0
                  </h1>
                  <h6
                    className="subtitle"
                  >
                    UserProfile.mots traduits
                  </h6>
                  <span
                    className="content texte-small"
                  >
                    UserProfile.commencez à traduire
                    .
                  </span>
                </AnchorLink>
              </Col>
            </Row>
          </CardBody>
        </Card>
      </Col>
    </Row>
    <withI18nextTranslation(favoriTable)
      headers={
        Array [
          <EVAIcon
            fill="#fff"
            name="bookmark"
            size="medium"
          />,
          "Titre",
          "Thèmes",
          "Ajouté le",
          "",
          "",
        ]
      }
      limit={5}
      removeBookmark={[Function]}
      title="Favoris"
      toggleModal={[Function]}
      upcoming={[Function]}
    />
    <withI18nextTranslation(contribTable)
      dataArray={Array []}
      deleteContrib={[Function]}
      displayIndicators={true}
      headers={
        Array [
          "Titre",
          "Statut",
          "Ils rédigent avec moi",
          "",
          "",
        ]
      }
      hide={false}
      hideOnPhone={
        Array [
          false,
          false,
          true,
          false,
          false,
        ]
      }
      limit={5}
      overlayBtn="Découvrir comment contribuer"
      overlayRedirect={false}
      overlaySpan="Réfugiés-info est une plateforme contributive, vous pouvez participer à son enrichissement"
      overlayTitle="Rédigez des nouveaux contenus"
      title="Rédactions"
      toggleModal={[Function]}
      toggleSection={[Function]}
      type="user"
      user={
        Object {
          "_id": "a0000000000000000000000a",
          "contributions": Array [],
          "cookies": Object {},
          "description": "description",
          "email": "mockUser@mockUser.fr",
          "roles": Array [],
          "selectedLanguages": Array [],
          "status": "Actif",
          "structures": Array [],
          "username": "mockUser",
        }
      }
    />
    <withI18nextTranslation(tradTable)
      dataArray={Array []}
      displayIndicators={true}
      headers={
        Array [
          "Titre",
          "Statut",
          "Progression",
          "Langue",
          "Ils rédigent avec moi",
          "",
        ]
      }
      hide={false}
      hideOnPhone={
        Array [
          false,
          false,
          true,
          false,
          true,
          false,
        ]
      }
      langues={Array []}
      limit={5}
      minutesPassees={0}
      motsRediges={0}
      overlayBtn="Démarrer une session"
      overlayRedirect={false}
      overlaySpan="Bilingue ? Polyglotte ? Participez à l’effort de traduction à votre rythme :"
      overlayTitle="Aidez à traduire les contenus"
      overlayi18n="bilingue"
      title="Traductions"
      toggleModal={[Function]}
      toggleSection={[Function]}
      user={
        Object {
          "_id": "a0000000000000000000000a",
          "contributions": Array [],
          "cookies": Object {},
          "description": "description",
          "email": "mockUser@mockUser.fr",
          "roles": Array [],
          "selectedLanguages": Array [],
          "status": "Actif",
          "structures": Array [],
          "username": "mockUser",
        }
      }
    />
    <withI18nextTranslation(actionTable)
      archive={[Function]}
      dataArray={Array []}
      headers={
        Array [
          <EVAIcon
            fill="#fff"
            name="bell-outline"
            size="medium"
          />,
          "Titre",
          "Mon rôle",
          "Type",
          "Depuis",
          "",
          "",
        ]
      }
      hideOnPhone={
        Array [
          false,
          false,
          true,
          false,
          true,
          false,
          false,
        ]
      }
      limit={5}
      showSuggestion={[Function]}
      title="Notifications"
      toggleModal={[Function]}
      upcoming={[Function]}
    />
  </div>
  <Modal
    autoFocus={true}
    backdrop={true}
    backdropTransition={
      Object {
        "mountOnEnter": true,
        "timeout": 150,
      }
    }
    centered={false}
    className="modal-plus"
    fade={true}
    isOpen={false}
    keyboard={true}
    modalTransition={
      Object {
        "timeout": 300,
      }
    }
    onClosed={[Function]}
    onOpened={[Function]}
    role="dialog"
    toggle={[Function]}
    zIndex={1050}
  >
    <withI18nextTranslation(actionTable)
      archive={[Function]}
      dataArray={Array []}
      headers={
        Array [
          <EVAIcon
            fill="#fff"
            name="bell-outline"
            size="medium"
          />,
          "Titre",
          "Mon rôle",
          "Type",
          "Depuis",
          "",
          "",
        ]
      }
      hideOnPhone={
        Array [
          false,
          false,
          true,
          false,
          true,
          false,
          false,
        ]
      }
      showSuggestion={[Function]}
      title="Notifications"
      toggleModal={[Function]}
    />
  </Modal>
  <Modal
    autoFocus={true}
    backdrop={true}
    backdropTransition={
      Object {
        "mountOnEnter": true,
        "timeout": 150,
      }
    }
    centered={false}
    className="modal-plus"
    fade={true}
    isOpen={false}
    keyboard={true}
    modalTransition={
      Object {
        "timeout": 300,
      }
    }
    onClosed={[Function]}
    onOpened={[Function]}
    role="dialog"
    toggle={[Function]}
    zIndex={1050}
  >
    <withI18nextTranslation(contribTable)
      dataArray={Array []}
      deleteContrib={[Function]}
      headers={
        Array [
          "Titre",
          "Statut",
          "Ils rédigent avec moi",
          "",
          "",
        ]
      }
      hideOnPhone={
        Array [
          false,
          false,
          true,
          false,
          false,
        ]
      }
      title="Rédactions"
      toggleModal={[Function]}
      type="user"
      user={
        Object {
          "_id": "a0000000000000000000000a",
          "contributions": Array [],
          "cookies": Object {},
          "description": "description",
          "email": "mockUser@mockUser.fr",
          "roles": Array [],
          "selectedLanguages": Array [],
          "status": "Actif",
          "structures": Array [],
          "username": "mockUser",
        }
      }
    />
  </Modal>
  <Modal
    autoFocus={true}
    backdrop={true}
    backdropTransition={
      Object {
        "mountOnEnter": true,
        "timeout": 150,
      }
    }
    centered={false}
    className="modal-plus"
    fade={true}
    isOpen={false}
    keyboard={true}
    modalTransition={
      Object {
        "timeout": 300,
      }
    }
    onClosed={[Function]}
    onOpened={[Function]}
    role="dialog"
    toggle={[Function]}
    zIndex={1050}
  >
    <withI18nextTranslation(tradTable)
      dataArray={Array []}
      headers={
        Array [
          "Titre",
          "Statut",
          "Progression",
          "Langue",
          "Ils rédigent avec moi",
          "",
        ]
      }
      hideOnPhone={
        Array [
          false,
          false,
          true,
          false,
          true,
          false,
        ]
      }
      langues={Array []}
      title="Traductions"
      toggleModal={[Function]}
      user={
        Object {
          "_id": "a0000000000000000000000a",
          "contributions": Array [],
          "cookies": Object {},
          "description": "description",
          "email": "mockUser@mockUser.fr",
          "roles": Array [],
          "selectedLanguages": Array [],
          "status": "Actif",
          "structures": Array [],
          "username": "mockUser",
        }
      }
    />
  </Modal>
  <Modal
    autoFocus={true}
    backdrop={true}
    backdropTransition={
      Object {
        "mountOnEnter": true,
        "timeout": 150,
      }
    }
    centered={false}
    className="modal-plus"
    fade={true}
    isOpen={false}
    keyboard={true}
    modalTransition={
      Object {
        "timeout": 300,
      }
    }
    onClosed={[Function]}
    onOpened={[Function]}
    role="dialog"
    toggle={[Function]}
    zIndex={1050}
  >
    <withI18nextTranslation(favoriTable)
      headers={
        Array [
          <EVAIcon
            fill="#fff"
            name="bookmark"
            size="medium"
          />,
          "Titre",
          "Thèmes",
          "Ajouté le",
          "",
          "",
        ]
      }
      removeBookmark={[Function]}
      title="Favoris"
      toggleModal={[Function]}
    />
  </Modal>
  <thanksModal
    show={false}
    toggle={[Function]}
  />
  <suggestionModal
    show={false}
    suggestion={Object {}}
    toggle={[Function]}
  />
  <withRouter(withI18nextTranslation(Connect(TraducteurModal)))
    langues={Array []}
    redirect={true}
    show={false}
    toggle={[Function]}
    user={
      Object {
        "_id": "a0000000000000000000000a",
        "contributions": Array [],
        "cookies": Object {},
        "description": "description",
        "email": "mockUser@mockUser.fr",
        "roles": Array [],
        "selectedLanguages": Array [],
        "status": "Actif",
        "structures": Array [],
        "username": "mockUser",
      }
    }
  />
  <ObjectifsModal
    show={false}
    toggle={[Function]}
    validateObjectifs={[Function]}
  />
  <addMemberModal
    addMember={[Function]}
    selectItem={[Function]}
    selected={Object {}}
    show={false}
    toggle={[Function]}
    users={Array []}
  />
  <PasswordModal
    changePassword={[Function]}
    cpassword=""
    newPassword=""
    onChange={[Function]}
    onClick={[Function]}
    password=""
    passwordVisible={false}
    show={false}
    t={[Function]}
    toggle={[Function]}
  />
</div>
`;
